Core/RestaurantReservation.Application/Contracts/IRepository/ICustomerRepository.cs

```diff
@@ -0,0 +1,9 @@
+﻿using RestaurantReservation.Application.DTOs;
+
+namespace RestaurantReservation.Application.Contracts.IRepository
+{
+    public interface ICustomerRepository : IEntityRepository<CustomerDTO>
+    {
+        Task<IEnumerable<CustomerDTO>> FindCustomersByPartySizeAsync(int partySize, int pageNumber, int pageSize);
```

Comment:
Check out how to use seive to implement such search behaviour, where we can make it easier to filter on more fields/conditions.